libs/mpi/CMakeLists.txt

```diff
@@ -0,0 +1,101 @@
+# Copyright (c) 2019 The STE||AR-Group
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+cmake_minimum_required(VERSION 3.3.2 FATAL_ERROR)
+
+# if networking is off, or MPI parcelport is not being used
+# then allow the option of using an MPI
+if ("${HPX_WITH_NETWORKING}" AND "${HPX_WITH_PARCELPORT_MPI}")
+  hpx_option(HPX_WITH_MPI_FUTURES BOOL
+    "Enable support for returning futures from MPI asynchronous calls"
+    " (default: ON if networking+MPI already enabled)"
+    ON CATEGORY "MPI")
+else()
+  hpx_option(HPX_WITH_MPI_FUTURES BOOL
+    "Enable support for returning futures from MPI asynchronous calls"
+    " (default: ON if networking+MPI already enabled)"
+    OFF CATEGORY "MPI")
+endif()
+
+# if the user does not want mpi async futures, quit - the module will not be enabled
+if (NOT ${HPX_WITH_MPI_FUTURES})
+  return()
+endif()
+
+#
+# only process further when enabled
+#
+list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+
+find_package(MPI REQUIRED)
+
+if(${CMAKE_VERSION} VERSION_LESS "3.12.0")
+    message(ERROR "cmake 3.12 or newer required")
+endif()
+
+# All cmake version don't have the same found variable set
+if(NOT MPI_FOUND AND NOT MPI_C_FOUND)
+  hpx_error("MPI could not be found and hpx::mpi library was requested\n
+  set MPI_MPI_CXX_COMPILER to point to your MPI installation's mpicxx")
```

Comment:
This error message is misleading. Finding the MPI libraries is not related to finding mpicxx.